File: src/test/java/hello/hellospring/service/MemberServiceTest.java
package hello.hellospring.service;

import hello.hellospring.domain.Member;
import org.assertj.core.api.Assertions;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class MemberServiceTest {

    MemberService memberService = new MemberService();

    @Test
    void 회원가입() {
        // Test는 한글로 이름을 적어도 상관없음.

        // given : 뭔가가 주어졌을 때
        Member member = new Member();
        member.setName("hello");

        // when : 이걸 실행했을 때
        Long saveId = memberService.join(member);

        // then : 결과가 이게 나와야 함
        Member findMember = memberService.findOne(saveId).get();
        Assertions.assertThat(member.getName()).isEqualTo(findMember.getName());

    }

    @Test
    public void 중복_회원_예외() {
        // given
        Member member1 = new Member();
        member1.setName("spring");

        Member member2 = new Member();
        member2.setName("spring");

        // when
        memberService


        // then
    }

    @Test
    void findMembers() {
    }

    @Test
    void findOne() {
    }
}

File: src/test/kotlin/hello/hellospring/service/MemberServiceTest.kt
package hello.hellospring.service

import hello.hellospring.domain.Member
import hello.hellospring.dto.MemberDto
import hello.hellospring.repository.MemoryMemberRepository
import org.assertj.core.api.Assertions
import org.assertj.core.api.Assertions.*
import org.junit.jupiter.api.AfterEach
import org.junit.jupiter.api.Test

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.DisplayName
import org.springframework.beans.factory.annotation.Autowired

class MemberServiceTest(
    private val memberService: MemberService,
    private val memberRepository: MemoryMemberRepository
) {

    @BeforeEach
    fun beforeEach() {
        val memberService = MemberService(memberRepository)
    }

    @AfterEach
    fun afterEach() {
        memberRepository.clearStore()
    }

    @Test
    @DisplayName("회원가입")
    fun join() {
        val member = MemberDto("spring", null)

        memberService.join(member)

        val results = memberRepository.findAll()
        assertThat(results).hasSize(1)
        assertThat(results[0].name).isEqualTo("spring")
        assertThat(results[0].id).isNull()
    }

    @Test
    fun validateDuplicateMember() {
    }

    @Test
    fun findMembers() {
    }

    @Test
    fun findOne() {
    }
}